src/testrand_impl.h

```diff
@@ -14,22 +14,44 @@
 #include "testrand.h"
 #include "hash.h"
 
-static secp256k1_rfc6979_hmac_sha256 secp256k1_test_rng;
-static uint32_t secp256k1_test_rng_precomputed[8];
-static int secp256k1_test_rng_precomputed_used = 8;
+static uint64_t secp256k1_test_rng_key1, secp256k1_test_rng_key2;
+static uint64_t secp256k1_test_rng_cnt = 1;
 static uint64_t secp256k1_test_rng_integer;
 static int secp256k1_test_rng_integer_bits_left = 0;
 
-SECP256K1_INLINE static void secp256k1_testrand_seed(const unsigned char *seed16) {
-    secp256k1_rfc6979_hmac_sha256_initialize(&secp256k1_test_rng, seed16, 16);
+SECP256K1_INLINE static uint32_t secp256k1_testrand32(void) {
+    /* RNG based on https://arxiv.org/abs/2004.06278, using two separate keys, and
+     * only using odd counters to avoid entropy loss of key1. */
```

Comment:
It may be worth stressing in the comment that it's for tests only, for someone who runs `grep RNG`...

Reply:
Done.